internal/util/topology_test.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package util
+package util // nolint:testpackage // we're testing internals of the package.
```

Comment:
In some places the package name is modified in some places its not. May I know why?

Reply:
Whichever places we could address the warnings(test packages should be renamed *_test) by not exposing internal functionality, the changes are made there. nolint otherwise.

Reply:
see here

https://github.com/ceph/ceph-csi/pull/1157/files#diff-6f0b3665bd25cc9d4bf0a6d78505071eR17
https://github.com/ceph/ceph-csi/pull/1157/files#diff-4fc5079c0839e9840112be8ddec44e31R17
https://github.com/ceph/ceph-csi/pull/1157/files#diff-3f9b851a3d8cf4f5031c78712c15e5a4R17
in this places its renamed to util_test.

Reply:
By renaming test packages to `*_test` only public functions/members can be used while testing. This is for doing API tests in black-box type of tests.

At the moment, many packages  provide too much functionality to actually do sufficient black-box testing. We need to test internals of most (if not all) packages.

I am not convinced renaming packages to `*_test` is appropriate. Disabling the `testpackage` linter is done in one of the commits, so this should not get reported anymore?

Reply:
These test packages whose name is now changed to `util_test`, were the packages that didn't access any internal functionalities of `util`. They only made use of Exported parameters, thus they were put in a new package name `util_test` (To solve the warning `package should be `util_test` instead of `util` (testpackage)`). 
On the other hand, the test packages under util that used internal functionality, their package name was not changed (and just nolinted).

Reply:
Should we disable all testpackages?

Reply:
Yes disabling would be the best option for now

Reply:
Sure, will do it. Thanks.